tests/Unit/Evolution/DgSubcell/Test_Matrices.cpp

```diff
@@ -0,0 +1,168 @@
+#include "Evolution/DgSubcell/Matrices.hpp"
+#include "Helpers/Evolution/DgSubcell/ProjectionTestHelpers.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Blas.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeArray.hpp"
+
+namespace evolution::dg::subcell::fd {
+namespace {
+template <size_t MaxPts, size_t Dim, Spectral::Basis BasisType,
+          Spectral::Quadrature QuadratureType>
+void test_projection_matrix() noexcept {
+  CAPTURE(Dim);
+  CAPTURE(BasisType);
+  CAPTURE(QuadratureType);
+  static_assert(Spectral::maximum_number_of_points<BasisType> >= MaxPts,
+                "Basis does not have enough points. Must have at least 8 "
+                "points per element");
+
+  for (size_t num_pts_1d = std::max(
+           static_cast<size_t>(2),
+           Spectral::minimum_number_of_points<BasisType, QuadratureType>);
+       num_pts_1d < MaxPts + 1; ++num_pts_1d) {
+    const Mesh<Dim> dg_mesh{num_pts_1d, BasisType, QuadratureType};
+    const auto logical_coords = logical_coordinates(dg_mesh);
+    const size_t num_subcells_1d = 2 * num_pts_1d - 1;
+    const Mesh<Dim> subcell_mesh(num_subcells_1d,
+                                 Spectral::Basis::FiniteDifference,
+                                 Spectral::Quadrature::CellCentered);
+    const size_t num_subcells = subcell_mesh.number_of_grid_points();
+    const DataVector nodal_coeffs =
+        TestHelpers::evolution::dg::subcell::cell_values(dg_mesh,
+                                                         logical_coords);
+
+    const Matrix& proj_matrix =
+        projection_matrix(dg_mesh, subcell_mesh.extents());
+
+    DataVector cell_centered_values(num_subcells, 0.0);
+    dgemv_('N', proj_matrix.rows(), proj_matrix.columns(), 1.0,
+           proj_matrix.data(), proj_matrix.rows(), nodal_coeffs.data(), 1, 0.0,
+           cell_centered_values.data(), 1);
+
+    CHECK_ITERABLE_APPROX(cell_centered_values,
+                          TestHelpers::evolution::dg::subcell::cell_values(
+                              dg_mesh, logical_coordinates(subcell_mesh)));
+  }
+}
+
+template <size_t MaxPts, size_t Dim, Spectral::Basis BasisType,
+          Spectral::Quadrature QuadratureType>
+void reconstruction_matrix(const double eps) noexcept {
+  CAPTURE(Dim);
+  CAPTURE(BasisType);
+  CAPTURE(QuadratureType);
+  static_assert(Spectral::maximum_number_of_points<BasisType> >= MaxPts,
+                "Basis does not have enough points. Must have at least 8 "
+                "points per element");
+  Approx local_approx = Approx::custom().epsilon(eps).scale(1.);
+
+  for (size_t num_pts_1d = std::max(
+           static_cast<size_t>(2),
+           Spectral::minimum_number_of_points<BasisType, QuadratureType>);
+       num_pts_1d < MaxPts + 1; ++num_pts_1d) {
+    CAPTURE(MaxPts);
+    CAPTURE(Dim);
+    CAPTURE(BasisType);
+    CAPTURE(QuadratureType);
+    CAPTURE(num_pts_1d);
```

Comment:
I think you can remove the captures here that are also just above (outside the loop)